src/layergroup.js

```diff
@@ -0,0 +1,176 @@
+   * @property {number} miny The minimum y coordinates. Default: -90
+   * @property {number} maxx The maximum x coordinates. Default: 180
+   * @property {number} maxy The maximum y coordinates. Default: 90
+   * @property {String} crs The crs of the bounds. Default: 'EPSG:4326'
+   */
+
+  /**
+   * Create a GeoServer layergroup by the given workspace, layerGroupName, layers and options
+   * @param {String} workspace The name of the workspace
+   * @param {String} layerGroupName The name of the layer group
+   * @param {String} layerGroupTitle The title of the layer group
+   * @param {Array.<String>} layers List of layers to be added to the group. Must be in same workspace as layergroup
+   * @param {String} options.mode The mode of the layergroup. Default to SINGLE
+   * @param {String} options.layerGroupTitle The title of the layergroup.
+   * @param {bounds} options.bounds The bounds of the layer group.
+   *
+   * @throws Error if request fails
+   *
+   * @returns {Object} A string with layer group location or undefined if not found
+   */
+  async create (workspace, layerGroupName, layers, layerGroupOptions) {
+    const options = {
+      mode: 'SINGLE',
+      layerGroupTitle: '',
+      abstractTxt: '',
+      bounds: {
+        minx: -180,
+        maxx: -90,
+        miny: 180,
+        maxy: 90,
+        crs: 'EPSG:4326'
+      },
+      ...layerGroupOptions
+    };
+    const publishedLayers = [];
+    const styles = [];
+    for (const l of layers) {
+      publishedLayers.push({
+        '@type': 'layer',
+        name: `${workspace}:${l}`,
+        href: `${this.url}/workspaces/${workspace}/layers/${l}.json`
+      });
+      // use default style by define empty string
+      styles.push('');
+    }
+    const body = {
+      layerGroup: {
+        name: layerGroupName,
+        workspace: {
+          name: workspace
+        },
+        publishables: {
+          published: publishedLayers
+        },
+        styles: {
+          style: styles
+        },
+        ...options
+      }
+    };
+    const response = await fetch(
+      `${this.url}/workspaces/${workspace}/layergroups`, {
+        credentials: 'include',
+        method: 'POST',
+        headers: {
+          Authorization: this.auth,
+          'Content-Type': 'application/json'
+        },
+        body: JSON.stringify(body)
+      }
+    );
+
+    if (!response.ok) {
+      const grc = new AboutClient(this.url, this.auth);
+      if (await grc.exists()) {
+        // GeoServer exists, but requested item does not exist, we return empty
+        return;
+      } else {
+        // There was a general problem with GeoServer
+        const geoServerResponse = await getGeoServerResponseText(response);
+        throw new GeoServerResponseError(null, geoServerResponse);
+      }
+    }
+    // return resource location in header
+    const location = response.headers.get('location');
+
+    return location;
```

Comment:
Could you give brief info, what exactly is returned here?

Maybe we should harmonize the return values especially for the create functions of the different clients in general. But unrelated to this PR.

Reply:
See https://github.com/meggsimum/geoserver-node-client/issues/143

Reply:
good point!